observability-kit-starter/src/main/java/com/vaadin/observability/ObservabilityHandler.java

```diff
@@ -89,31 +138,63 @@ public boolean synchronizedHandleRequest(VaadinSession session,
         }
 
         try {
-            ObjectMapper mapper = new ObjectMapper();
-            JsonNode root = mapper.readTree(request.getInputStream());
-
-            if (!root.has("resourceSpans")) {
+            Map<String,Object> objectMap =
+                    new ObjectMapper().readerForMapOf(Object.class)
+                            .readValue(request.getInputStream());
+            if (!objectMap.containsKey("resourceSpans")) {
                 getLogger().error("Malformed traces message.");
                 response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
                 return true;
             }
-
-            handleTraces(root);
-
+            exporter.accept(id, objectMap);
             response.setStatus(HttpServletResponse.SC_OK);
         } catch (IOException e) {
-            getLogger().error("Exception when processing traces.");
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
         }
 
         return true;
     }
 
+    /**
+     * Returns the unique ID for the installed handler.
+     *
+     * @return the unique handler ID
+     */
     public String getId() {
         return id;
     }
 
-    private void handleTraces(JsonNode root) {
-        // Dummy method for the agent to hook onto
+    /**
+     * Gets the string value of the provided configuration key name from a
+     * callback injected by the Observability agent.
+     *
+     * @param key
+     *            the configuration key name
+     * @return the string value of the key
+     */
+    public String getConfigProperty(String key) {
+        return config.apply(key);
+    }
+
+    /**
+     * Override default deserialization logic to account for transient fields.
+     *
+     * @param stream
+     *            the object to read
+     * @throws IOException
+     *            if an IO error occured
+     * @throws ClassNotFoundException
+     *            if the class of the stream object could not be found
+     */
+    public void readObject(ObjectInputStream stream)
+            throws IOException, ClassNotFoundException {
+        stream.defaultReadObject();
+        config = (key) -> {
+            getLogger().error("");
+            return null;
+        };
+        exporter = (id, map) -> {
+            getLogger().error("");
+        };
```

Comment:
This will always overwrite fields injected by the advice.
In addition, probably during deserialization the constructor advice is not invoked.
Perhaps we can add the advice used for the constructor also to this method, so the above will be overwritten by the advice if the class is instrumented